core/codec/cbor/cbor.hpp

```diff
@@ -32,7 +36,7 @@ namespace fc::codec::cbor {
    * @see cbor_errors.hpp for possible error cases
    */
   template <typename T>
-  outcome::result<T> decode(const std::vector<uint8_t> &input) {
+  outcome::result<T> decode(gsl::span<const uint8_t> input) {
```

Comment:
How useful is it? Because now the signature of ```decode``` is not anymore an inverse of ```encode```:  ```std::vector<uint8_t> -> T | error``` (even though the isomorphic property ```decode(encode(a)) == a``` will still hold).

Reply:
span is better